File: dotnet-core-api/SplitExpense.SharedResource/ApiPaths.cs
namespace SplitExpense.SharedResource
{
    public static class ApiPaths
    {
        public const string ApiPrefix = "api/v1/[controller]/";
        #region Group APIs
        public const string GroupCreate = ApiPrefix + "create";
        public const string GroupGetAll = ApiPrefix + "get/all";
        public const string GroupGetRecent = ApiPrefix + "get/recent";
        public const string GroupGetById = ApiPrefix + "get/{id}";
        public const string GroupUpdate = ApiPrefix + "update";
        public const string GroupDelete = ApiPrefix + "delete/{id}";
        public const string GroupSearch = ApiPrefix + "search";
        public const string GroupAddUser = ApiPrefix + "add/friend";
        #endregion

        #region SplitType APIs
        public const string SplitTypeCreate = ApiPrefix + "create";
        public const string SplitTypeGetAll = ApiPrefix + "get/all";
        public const string SplitTypeGetById = ApiPrefix + "get/{id}";
        public const string SplitTypeUpdate = ApiPrefix + "update";
        public const string SplitTypeDelete = ApiPrefix + "delete/{id}";
        public const string SplitTypeSearch = ApiPrefix + "search";
        #endregion


        #region Email APIs
        public const string EmailGetQueue = ApiPrefix + "get/queue";
        #endregion

        #region Expense API Path
            public const string ExpenseCreate = ApiPrefix + "create";
            public const string ExpenseUpdate = ApiPrefix + "update";
            public const string ExpenseDelete = ApiPrefix + "delete/{id}";
            public const string ExpenseGetAll = ApiPrefix + "get/all";
            public const string ExpenseSearch = ApiPrefix + "search";
        #endregion

        #region Contact API Path
        public const string ContactCreate = ApiPrefix + "create";
        public const string ContactAddInList = ApiPrefix + "add/in/contact/list/{id}";
        public const string ContactGetAll = ApiPrefix + "get/all";
        public const string ContactGetById = ApiPrefix + "get/{id}";
        public const string ContactUpdate = ApiPrefix + "update";
        public const string ContactDelete = ApiPrefix + "delete/{id}";
        public const string ContactSearch = ApiPrefix + "search";
        public const string ContactSearchUser = ApiPrefix + "search/user";
        #endregion

        #region GroupType
        public const string GroupTypeCreate = ApiPrefix + "create";
        public const string GroupTypeUpdate = ApiPrefix + "update";
        public const string GroupTypeDelete = ApiPrefix + "delete/{id}";
        public const string GroupTypeGetById = ApiPrefix + "get/{id}";
        public const string GroupTypeGetAll = ApiPrefix + "get/all";
        public const string GroupTypeSearch = ApiPrefix + "search";
        #endregion

        #region Auth
        public const string Register = ApiPrefix + "register";
        public const string Login = ApiPrefix + "login";
        public const string Logout = ApiPrefix + "logout";
        public const string ForgotPassword = ApiPrefix + "forgot-password";
        public const string ResetPassword = ApiPrefix + "reset-password";
        public const string ForgotUsername = ApiPrefix + "forgot-username";
        public const string UpdateUserProfile = ApiPrefix + "update/profile";
        public const string UpdateUserProfilePicture = ApiPrefix + "update/profile-picture";
        public const string DeleteUserProfilePicture = ApiPrefix + "delete/profile-picture";
        public const string DeleteUserProfile = ApiPrefix + "delete/profile";
        #endregion

        #region ExpenseActivity
        public const string ExpenseActivityCreate = ApiPrefix + "create";
        public const string ExpenseActivityGetAll = ApiPrefix + "get/all";
        public const string ExpenseActivitySearch = ApiPrefix + "search";
        #endregion

        #region Email Template API
        public const string EmailTemplateCreate = ApiPrefix + "create";
        public const string EmailTemplateGetAll = ApiPrefix + "get/all";
        public const string EmailTemplateSearch = ApiPrefix + "search";
        public const string EmailTemplateUpdate = ApiPrefix + "update";
        public const string EmailTemplateDelete = ApiPrefix + "delete/{id}";
        #endregion
    }
}

File: react-app/split-expense-web/src/constants/apiPaths.js
import env from '../config/env.config';

export const API_BASE_URL = env.API_URL;

export const GROUP_PATHS = {
    CREATE: `${API_BASE_URL}/group/create`,
    LIST: `${API_BASE_URL}/group/get/all`,
    GET_RECENTS: `${API_BASE_URL}/group/get/recent`,
    UPDATE: `${API_BASE_URL}/group/update`,
    DELETE: `${API_BASE_URL}/group/delete`,
    SEARCH: `${API_BASE_URL}/group/search`,
    MEMBERS: `${API_BASE_URL}/contacts/get/all?pageno=1&pageSize=500`
};

export const GROUP_TYPES_PATHS={
    CREATE: `${API_BASE_URL}/grouptype/create`,
    UPDATE: `${API_BASE_URL}/grouptype/update`,
    DELETE: `${API_BASE_URL}/grouptype/delete/{id}`,
    GET_BY_ID: `${API_BASE_URL}/grouptype/{id}`,
    LIST: `${API_BASE_URL}/grouptype/get/all`,
    SEARCH: `${API_BASE_URL}/grouptype/search`
}

export const AUTH_PATHS = {
    LOGIN: `${API_BASE_URL}/auth/login`,
    UPDATE_PROFILE: `${API_BASE_URL}/auth/update`,
    LOGOUT: `${API_BASE_URL}/auth/logout`,
    REGISTER: `${API_BASE_URL}/auth/register`,
    GOOGLE_LOGIN: `${API_BASE_URL}/auth/google`,
    FORGOT_PASSWORD: `${API_BASE_URL}/auth/forgot-password`,
    FORGOT_USERNAME: `${API_BASE_URL}/auth/forgot-username`,
    RESET_PASSWORD: `${API_BASE_URL}/auth/reset-password`,
};

export const CONTACT_PATHS = {
    LIST: `${API_BASE_URL}/contacts/get/all`,
    SEARCH: `${API_BASE_URL}/contacts/search`,
    SEARCH_USER: `${API_BASE_URL}/contacts/search/user`,
    DELETE: `${API_BASE_URL}/contacts/delete`,
    ADD: `${API_BASE_URL}/contacts/add`,
    CREATE: `${API_BASE_URL}/contacts/create`,
    ADD_IN_CONTACT_LIST:`${API_BASE_URL}/contacts/add/in/contact/list/`,
}; 